tests/visualization_tests/test_param_importances.py

```diff
@@ -9,76 +13,121 @@
+parametrize_plot_param_importances = pytest.mark.parametrize(
+    "plot_param_importances", [plotly_plot_param_importances, plt_plot_param_importances]
+)
+
 
+def _create_study_with_failed_trial() -> Study:
+    study = create_study()
+    study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
+    return study
+
+
+def test_target_is_none_and_study_is_multi_obj() -> None:
     study = create_study(directions=["minimize", "minimize"])
     with pytest.raises(ValueError):
-        plot_param_importances(study)
+        _get_importances_info(
+            study=study, evaluator=None, params=None, target=None, target_name="Objective Value"
+        )
 
 
-def test_plot_param_importances() -> None:
+def test_get_param_importances_info_for_one_trial() -> None:
+    # Cannot obtain importance scores from one complete trial.
 
-    # Test with no trial.
     study = create_study()
-    figure = plot_param_importances(study)
-    assert len(figure.data) == 0
-
-    study = prepare_study_with_trials(with_c_d=True)
-
-    # Test with a trial.
-    figure = plot_param_importances(study)
-    assert len(figure.data) == 1
-    assert set(figure.data[0].y) == set(
-        ("param_b", "param_d")
-    )  # "param_a", "param_c" are conditional.
-    assert math.isclose(1.0, sum(i for i in figure.data[0].x), abs_tol=1e-5)
-    assert figure.layout.xaxis.title.text == "Importance for Objective Value"
-
-    # Test with an evaluator.
-    plot_param_importances(study, evaluator=MeanDecreaseImpurityImportanceEvaluator())
-    assert len(figure.data) == 1
-    assert set(figure.data[0].y) == set(
-        ("param_b", "param_d")
-    )  # "param_a", "param_c" are conditional.
-    assert math.isclose(1.0, sum(i for i in figure.data[0].x), abs_tol=1e-5)
-
-    # Test with a trial to select parameter.
-    figure = plot_param_importances(study, params=["param_b"])
-    assert len(figure.data) == 1
-    assert figure.data[0].y == ("param_b",)
-    assert math.isclose(1.0, sum(i for i in figure.data[0].x), abs_tol=1e-5)
-
-    # Test with a customized target value.
-    with pytest.warns(UserWarning):
-        figure = plot_param_importances(
-            study, target=lambda t: t.params["param_b"] + t.params["param_d"]
-        )
-    assert len(figure.data) == 1
-    assert set(figure.data[0].y) == set(
-        ("param_b", "param_d")
-    )  # "param_a", "param_c" are conditional.
-    assert math.isclose(1.0, sum(i for i in figure.data[0].x), abs_tol=1e-5)
-
-    # Test with a customized target name.
-    figure = plot_param_importances(study, target_name="Target Name")
-    assert figure.layout.xaxis.title.text == "Importance for Target Name"
+    study.optimize(lambda t: t.suggest_float("x", 0, 1), n_trials=1)
 
-    # Test with wrong parameters.
     with pytest.raises(ValueError):
-        plot_param_importances(study, params=["optuna"])
+        _get_importances_info(study, None, None, None, "Objective Value")
```

Comment:
This is covered in the tests of `get_param_importances`, so I think we don't need to check it here.

Reply:
Right. Thanks.